### PR TITLE
Azure single server to flexible server

### PR DESCRIPTION
**Pull request by tim-allen-ck**
# https://github.com/microsoft/AzureTRE/issues/3540

## What is being addressed

Single Server is retiring in favour of Flexible server so updating azurerm_mysql_server to azurerm_mysql_flexible_server and azurerm_mysql_database to azurerm_mysql_flexible_database

## How is this addressed

Update terraform files referencing azurerm_mysql_server to azurerm_mysql_flexible_server.
Update terraform files referencing azurerm_mysql_database to azurerm_mysql_flexible_database
sku_name changes
updates to GITEA__database__USER to use just the db username

### CODE REVIEW

**Comment by tim-allen-ck** [comment]
@microsoft-github-policy-service agree company="Cloudkubed"

**Comment by tim-allen-ck** [comment]
@microsoft-github-policy-service agree company="CloudKubed"

File: README.md
Comment (marrobi) [inline README.md:1]: Not sure this needs to be part of this PR?

**Review by marrobi — APPROVED** [review]
LGTM. Thanks!

File: .markdownlint.json [hunk 1]
```diff
@@ -13,4 +13,4 @@
   "MD033": false,
   "MD036": false,
   "blank_lines": false
-}
+}
```
Comment (tamirkamara) [inline .markdownlint.json:16]: Please add a new line

File: templates/shared_services/gitea/template_schema.json [hunk 2]
```diff
@@ -149,4 +162,4 @@
       }
     ]
   }
-}
+}
```
Comment (tamirkamara) [inline templates/shared_services/gitea/template_schema.json:165]: Please add a new line

File: templates/workspace_services/gitea/parameters.json [hunk 3]
```diff
@@ -63,6 +63,12 @@
...
+    },
+    {
+      "name": "sql_sku",
+      "source": {
+        "env": "SQL_SKU"
+      }
     }
   ]
-}
+}
```
Comment (tamirkamara) [inline templates/workspace_services/gitea/parameters.json:74]: Please add a new line

**Review by wojciechcloudkubed — APPROVED** [review]

**Review by tamirkamara — APPROVED** [review]

**Pull request merged**